lean4-infoview/src/infoview/main.tsx

```diff
@@ -52,11 +52,16 @@ function Main(props: {}) {
         []
     );
 
+    const serverInitializeResult = useEventResult(ec.events.serverRestarted);
```

Comment:
If `useEventResult` were higher-order and taking a function that maps the event value, we could also use it for `uri`/`events.changedCursorLocation` above.

Reply:
Good suggestion!  But there's no higher-order argument necessary, you can just do a `?.uri` afterwards.

Reply:
Added the higher-order version anyhow to reduce rerenders.